extension/parquet/parquet_reader.cpp

```diff
@@ -484,23 +479,33 @@ ParquetColumnDefinition ParquetColumnDefinition::FromSchemaValue(ClientContext &
 ParquetReader::ParquetReader(ClientContext &context_p, string file_name_p, ParquetOptions parquet_options_p)
     : fs(FileSystem::GetFileSystem(context_p)), allocator(BufferAllocator::Get(context_p)),
       parquet_options(std::move(parquet_options_p)) {
+
 	file_name = std::move(file_name_p);
 	file_handle = fs.OpenFile(file_name, FileFlags::FILE_FLAGS_READ);
 	if (!file_handle->CanSeek()) {
 		throw NotImplementedException(
 		    "Reading parquet files from a FIFO stream is not supported and cannot be efficiently supported since "
 		    "metadata is located at the end of the file. Write the stream to disk first and read from there instead.");
 	}
+
+	// set pointer to factory method for AES state
+	auto &config = DBConfig::GetConfig(context_p);
+	if (config.encryption_state && parquet_options.debug_use_openssl) {
```

Comment:
I wonder if we should use a new ClientContextState for this instead of putting this in the DBConfig?

Reply:
Initially I thought about something like that too. But what happens when we have multiple clients? Will this then still work?

Reply:
Yea you're right, I believe the `encryption_state` is read-only though, so perhaps there could be an object in the HTTPFS extension that is only referenced by the ClientContextStates?

But that sounds needlessly brittle and error prone, I was just thinking out loud, feel free to disregard this comment :)